Modules/Server/Sources/PocketCastsServer/Public/Cache/ShowInfoDataRetriever.swift

```diff
@@ -20,14 +16,8 @@ public actor ShowInfoDataRetriever {
         let url = ServerHelper.asUrl(ServerConstants.Urls.cache() + "mobile/show_notes/full/\(podcastUuid)")
         let request = URLRequest(url: url, cachePolicy: .reloadRevalidatingCacheData)
 
-        if let cachedResponse = showNotesUrlCache.cachedResponse(for: request) {
```

Comment:
Do you still only redownload the content if the last modified or etag have changed? Below is an example of the call. It calls the server to get the static immutable file location. Ideally you would only process this final json `/show_notes/e7a6f7d0-02f2-0133-1c51-059c869cc4eb/episodes_1712289890.json` once to save on downloads and client side processing. 
```
wget -S https://cache.pocketcasts.com/mobile/show_notes/full/e7a6f7d0-02f2-0133-1c51-059c869cc4eb
--2024-04-09 07:06:13--  https://cache.pocketcasts.com/mobile/show_notes/full/e7a6f7d0-02f2-0133-1c51-059c869cc4eb
Resolving cache.pocketcasts.com (cache.pocketcasts.com)... 3.217.52.176, 3.217.9.18, 3.216.204.171
Connecting to cache.pocketcasts.com (cache.pocketcasts.com)|3.217.52.176|:443... connected.
HTTP request sent, awaiting response...
  HTTP/1.1 302 Found
  Date: Mon, 08 Apr 2024 21:36:14 GMT
  Content-Length: 0
  Connection: keep-alive
  vary: origin
  Location: https://shownotes.pocketcasts.com/show_notes/e7a6f7d0-02f2-0133-1c51-059c869cc4eb/episodes_1712289890.json
Location: https://shownotes.pocketcasts.com/show_notes/e7a6f7d0-02f2-0133-1c51-059c869cc4eb/episodes_1712289890.json [following]
--2024-04-09 07:06:14--  https://shownotes.pocketcasts.com/show_notes/e7a6f7d0-02f2-0133-1c51-059c869cc4eb/episodes_1712289890.json
Resolving shownotes.pocketcasts.com (shownotes.pocketcasts.com)... 104.22.71.132, 104.22.70.132, 172.67.25.191
Connecting to shownotes.pocketcasts.com (shownotes.pocketcasts.com)|104.22.71.132|:443... connected.
HTTP request sent, awaiting response...
  HTTP/1.1 200 OK
  Date: Mon, 08 Apr 2024 21:36:15 GMT
  Content-Type: application/json; charset=utf-8
  Transfer-Encoding: chunked
  Connection: keep-alive
  x-amz-id-2: U/f3bixBrSIobmTIixduWbSTEVbLBMpsHSmJLoxr1PW6Z3ZqegMQq5Uh5HJvxjUPyx/t3r2vz/4=
  x-amz-request-id: 1V5WKBBQD6ACRH5V
  Last-Modified: Fri, 05 Apr 2024 04:04:52 GMT
  ETag: W/"8caf53682ef105234da3b896ab5382b6"
  x-amz-server-side-encryption: AES256
  CF-Cache-Status: DYNAMIC
  Server: cloudflare
  CF-RAY: 8715644ab90055f1-ADL
  ```

Reply:
We handle the cache ourselves now: https://github.com/Automattic/pocket-casts-ios/blob/trunk/podcasts/ShowInfoCoordinator.swift#L71

Client-side processing was our main goal, given that with the previous solution we had an increase in memory crashes due to the size of the JSON and `JSONDecoder` performance.

I removed the `URLCache` from here because it's not necessary anymore. :)

Reply:
I might be reading the Swift code wrong, but it only downloads the latest show notes file if you can't find an episode in your cache. Is that to reduce the amount the JSON is parsed? 
My only concern was that if any information changes about an older episode, the cache won't be updated, but I believe the previous implement did handle this using the `Last-Modified` header and 304 status code.

Reply:
You are right!

> Is that to reduce the amount the JSON is parsed?

This is to reduce the whole processing.

> My only concern was that if any information changes about an older episode, the cache won't be updated, but I believe the previous implement did handle this using the Last-Modified header and 304 status code.

How often does that happen?

I can change this behavior for `7.62` — for this version, it's a bit risky as it won't be that simple and we won't have enough time to test IMHO. How's Android handling that? Before it was straightforward because the show notes were displayed only in two places:

1. Episode detail
2. Full player

But now we also have episode artwork, and these can be displayed on any list. Is Pocket Casts Android requesting the endpoint whenever an episode artwork is shown?

Reply:
> But now we also have episode artwork, and these can be displayed on any list. Is Pocket Casts Android requesting the endpoint whenever an episode artwork is shown?

I would need to take a look when I'm back on Android. It feels like these calls should happen at the end of a refresh and only download the files that have changed, rather when you are navigating through the app. 🤔

Reply:
> It feels like these calls should happen at the end of a refresh and only download the files that have changed

I think this makes sense! Refresh and when loading a podcast, right?